drizzlepac/haputils/mvm_quality_analysis.py

```diff
@@ -247,6 +266,740 @@ def run_quality_analysis(total_obj_list,
+                        overlap_region_stats_descrip[stat_type] = stat_type
+                    else:
+                        overlap_region_stats_units[stat_type] = base_units
+                        overlap_region_stats_descrip[stat_type] = stat_type
+                # add "statistics" section of "stats_dict"
+                diagnostic_obj.add_data_item(stats_dict[bit_value]['statistics'][difference_type],
+                                             descriptions=overlap_region_stats_descrip,
+                                             units=overlap_region_stats_units,
+                                             title="{} {}".format(section_title_base_string, difference_type))
+
+            # build dictionary to populate "data table" section of json file
+            overlap_region_data_table_units = collections.OrderedDict()
+            overlap_region_data_table_descrips = collections.OrderedDict()
+            unit_list = ["HAP MVM skycell pixels",
+                         "HAP MVM skycell pixels",
+                         "Degrees",
+                         "Degrees",
+                         "HAP MVM skycell pixels",
+                         "HAP MVM skycell pixels",
+                         "HAP MVM skycell pixels",
+                         "Arcseconds"]
+            descrip_list = ["Crossmatched reference X coordinate",
+                            "Crossmatched reference Y coordinate",
+                            "Crossmatched reference right ascention coordinate",
+                            "Crossmatched reference declination coordinate",
+                            "Crossmatched comparison - reference X coordinate differences",
+                            "Crossmatched comparison - reference Y coordinate differences",
+                            "Crossmatched comparison - reference on-sky separation (X-Y)",
+                            "Crossmatched comparison - reference on-sky separation (RA-Dec)"]
+            for key_item, descrip_item, unit_item in zip(stats_dict[bit_value]["data table"].keys(), descrip_list, unit_list):
+                overlap_region_data_table_descrips[key_item] = descrip_item
+                overlap_region_data_table_units[key_item] = unit_item
+            # add "data table section" of "stats_dict"
+            diagnostic_obj.add_data_item(stats_dict[bit_value]["data table"],
+                                         descriptions=overlap_region_data_table_descrips,
+                                         units=overlap_region_data_table_units,
+                                         title="{} crossmatched reference X, Y, RA, Dec, and crossmatched comparison - reference difference values".format(section_title_base_string))
+
+        # Write out the file
+        diagnostic_obj.write_json_file(json_filename, clobber=True)
+        log.info("Wrote crossmatch results file {}".format(json_filename))
+
+        # Clean up
+        del diagnostic_obj
+
```

Comment:
I greatly appreciate that the logical steps for the main algorithm are numbered/organized for this procedural routine.  I also see that you have broken out major steps into their own functions.  Nevertheless, I find routines which are hundreds of lines long to be hard to follow, and I wonder if the code would benefit from a more modular approach.  Due to the constraints on your time, I am NOT suggesting you rework this code at this time.